packages/lisk-transactions/src/utils/verify.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import * as BigNum from 'browserify-bignum';
+import { TransactionError, TransactionPendingError } from '../errors';
+import { Account } from '../transaction_types';
+import { MultisignatureStatus } from '../transactions/base';
+import { convertBeddowsToLSK } from '../utils/format';
+import {
+	validateMultisignatures,
+	validateSignature,
+} from '../utils/sign_and_verify';
+
+export const verifySenderPublicKey = (
+	id: string,
+	sender: Account,
+	publicKey: string,
+): TransactionError | undefined =>
+	sender.publicKey !== publicKey
+		? new TransactionError('Invalid sender publicKey', id, '.senderPublicKey')
+		: undefined;
+
+export const verifySenderId = (
+	id: string,
+	sender: Account,
+	address: string,
+): TransactionError | undefined =>
+	sender.address.toUpperCase() !== address.toUpperCase()
+		? new TransactionError('Invalid sender address', id, '.senderId')
+		: undefined;
+
+export const verifyBalance = (
+	id: string,
+	sender: Account,
+	fee: BigNum,
+): TransactionError | undefined =>
+	new BigNum(sender.balance).lt(new BigNum(fee))
+		? new TransactionError(
+				`Account does not have enough LSK: ${
```

Comment:
We can make this function more extensive by changing the `fee` parameter name with more generalized name like `amount` so we can this function in transfer transaction as well.